AAv2-doc-updates (#3945)
mattermost/docs: source/deployment/auth.rst

..  _auth_mfa:

Multi-factor Authentication
===========================

Multi-factor authentication (MFA) is a secondary layer of user authentication applied to your Mattermost server.

> With MFA-enabled, users need to provide a secure one-time code generated by an app or smartphone in addition to their username and password in order to log in to Mattermost. MFA is useful in organizations that have specific security and compliance policies. It can also be used in organizations where Mattermost is not behind a firewall and doesn't have access to existing MFA infrastructure.

Reviewer: ```suggestion
With MFA enabled, users need to provide a secure one-time code in addition to their username and password in order to log in to Mattermost. MFA is useful in organizations that have specific security and compliance policies. It can also be used in organizations where Mattermost is not behind a firewall and doesn't have access to existing MFA infrastructure.
```

```suggestion
With MFA enabled, users provide a secure one-time code in addition to their username and password in order to log in to Mattermost. Some IT security policies require MFA, and it can improve security for organizations that choose not to secure Mattermost behind a firewall.
```

> Mattermost offers smartphone-based authentication in addition to email-password or Active Directory/LDAP authentication, to sign in to the Mattermost server.

Reviewer: ```suggestion
Mattermost offers a smartphone-based MFA check in addition to email-password or Active Directory/LDAP authentication to sign in to the Mattermost server.
```

Supported smartphones include iOS, Android, Blackberry, and Windows Phone devices that are able to install `Google Authenticator <https://support.google.com/accounts/answer/1066447?hl=en>`__. Other than requiring internet access to download and install Google Authenticator, the phone used for Mattermost MFA does not require internet access.

.. note::
  
  As the MFA implementation relies on Time-based One-time passwords (TOTP) ensure that your server system time is accurate. If there is a discrepancy, users may not be able to log in successfully.

Enabling MFA
------------

This option is enabled by the System Admin in the `System Console <http://docs.mattermost.com/administration/config-settings.html#mfa>`__ under **Authentication > MFA > Enable Multi-factor Authentication**. 

> Once enabled, users can set up this feature in the `Account Settings <https://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication>`__ menu under **Security > Multi-factor Authentication**.

Reviewer: ```suggestion
Once enabled, users can opt to use MFA on their account in the `Account Settings <https://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication>`__ menu under **Security > Multi-factor Authentication**.
```

> Once enabled, users can set up this feature in the `Account Settings <http://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication-enterprise>`__ menu under **Security > Multi-factor Authentication**.

Reviewer 2: ```suggestion
Once enabled, users can set up this feature in the `Account Settings <https://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication>`__ menu under **Security > Multi-factor Authentication**.
```

This option can be enabled by the System Admin in the `System Console <http://docs.mattermost.com/administration/config-settings.html#mfa>`__ under **Authentication > MFA > Enable Multi-factor Authentication**. End users can set up this feature in the `Account Settings <http://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication-enterprise>`__ menu under **Security > Multi-factor Authentication**.
This option is enabled by the System Admin in the `System Console <http://docs.mattermost.com/administration/config-settings.html#mfa>`__ under **Authentication > MFA > Enable Multi-factor Authentication**. 

Once enabled, users can set up this feature in the `Account Settings <https://docs.mattermost.com/help/settings/account-settings.html#multi-factor-authentication>`__ menu under **Security > Multi-factor Authentication**.

Enforcing MFA (E10)
-------------------

> This option can be enabled by the System Admin in the `System Console <http://docs.mattermost.com/administration/config-settings.html#mfa>`__ under **Authentication > MFA > Enforce Multi-factor Authentication**.

Reviewer: I'd leave this line in.
